src/contracts/Datatoken4.ts

```diff
@@ -30,19 +30,23 @@ export class Datatoken4 extends Datatoken {
     abi?: AbiItem[]
   ) {
     super(signer, network, config, abi)
+    this.abi = this.getDefaultAbi()
     // Wrap signer's address for encrypted data tx
     this.signer = sapphire.wrap(signer)
```

Comment:
is the template only for confidential EVM sapphire?
Cause we're wrapping the signer with their sdk call `sapphire.wrap(...)` ..
and we have a few functions doing it again like:
`confidentialEVM === true ? sapphire.wrap(this.signer) : this.signer, ...`
So, we get a "double" wrap() for confidential EVM and a wrap() for any other EVM using this template 
For now, i would just remove the wrap() call on the constructor, and later, once we merge the other PR we can refactor this a bit more (from the config we would know if is confidential evm and if we should wrap the calls

Reply:
I can remove the wrapper from constructor to avoid confusions.

Reply:
Template 4 is only for confidential EVM, I mean, the scope of its creation is destinated for sapphire and files encryption on sapphire chain.